packages/valory/skills/liquidity_trader_abci/rounds.py

```diff
@@ -395,6 +402,61 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         return None
 
 
+class DecideAgentStartingRound(VotingRound):
+    """DecisionMakingRound"""
+
+    payload_class = DecideAgentStartingPayload
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    no_majority_event = Event.NO_MAJORITY
+    collection_key = get_name(SynchronizedData.participant_to_votes)
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.positive_vote_threshold_reached:
+            # We reference all the events here to prevent the check-abciapp-specs tool from complaining
+            synchronized_data = cast(SynchronizedData, self.synchronized_data)
+
+            return synchronized_data, Event.MOVE_TO_NEXT_AGENT
+
+        if self.negative_vote_threshold_reached:
+            return self.synchronized_data, Event.DONT_MOVE_TO_NEXT_AGENT
+        if self.none_vote_threshold_reached:
+            return self.synchronized_data, self.none_event
+        if not self.is_majority_possible(
+            self.collection, self.synchronized_data.nb_participants
+        ):
+            return self.synchronized_data, self.no_majority_event
+        return None
```

Comment:
Why override the end block?

Reply:
Removed the overriding end block in Starting and Ending Round